DM-51490: Add guider ingest.

ktlim opened the pull request:
This is a simplistic version that sorts guiders to the end, ingests them using the method in `obs_lsst`, and places them back on the queue if they failed (so they can be retried hopefully after a non-guider image has appeared).

ktlim committed:
Ingest guiders.

ktlim committed:
Remove unused Rucio code.

ktlim committed:
Update stack versions.

ktlim committed:
Reorganize guider ingest.

ktlim committed:
Delay retry after guider failure.

ktlim committed:
Ignore webhook exceptions.

ktlim committed:
Add skip_existing.

This may help with double notifications.

ktlim committed:
Stop skipping existing, retry one by one.

ktlim committed:
Move retry.

ktlim committed:
Don't wait to retry ingests.

ktlim committed:
Add wait to other guider failure branch.

ktlim committed:
Add MAX_FAILURES env var.

ktlim committed:
Don't retry already-successful ingests.

ktlim committed:
Handle all files on ingest failure.

atanikan commented on src/ingest.py:
```diff
@@ -97,32 +88,77 @@ def on_success(datasets):
+        if "Datastore already contains" in str(exc):
+            # Don't retry these
+            r.lrem(worker_queue, 0, info.path)
+            return
```
Line 118. Change to continue. With my limited knowledge. The rest looks good to me

ktlim committed:
Rework the ingest error handling once more.

If any ingestion in a batch fails, none of the batch is ingested, so we need
to retry one-by-one, even for guiders.  (We might have a mix of old guiders
and new ones.)

The on_ingest_failure() callback is not useful for the batch, though, since
some of the files in the batch may actually be able to be ingested, so we only
use it during the one-by-one retries.

If there are any guiders that fail during the one-by-one retry, we delay the
next retry for 0.5 seconds to see if we can get a science image to define the
exposure record.

ktlim committed:
Don't worry about duplicate ingests.

ktlim committed:
Clean up guider wait-for-science.
